packages/broker/src/ConfigWizard.ts

```diff
@@ -82,200 +84,164 @@ const DefaultConfig: Config = {
+        },
+        validate: (input: string, answers: inquirer.Answers): string | boolean => {
+            const portNumber = parseInt(input || answers[`${pluginName}Port`])
+            if (!Number.isInteger(portNumber)) {
+                return `Non-numeric value provided`
             }
-
-            if (plugins[i] === 'HttpPublish') {
-                this.config.plugins['legacyPublishHttp'] = {}
+   
+            if (portNumber < MIN_PORT_VALUE || portNumber > MAX_PORT_VALUE) {
+                return `Out of range port ${portNumber} provided (valid range ${MIN_PORT_VALUE}-${MAX_PORT_VALUE})`
             }
+   
+            return true
+        },
+        default: plugin.port
+    }
+})
+
+prompts = prompts.concat(pluginSelectorPrompt).concat(pluginPrompts)
+
+export const getConfigFromAnswers = (answers: any): Config => {
+    const config = DEFAULT_CONFIG
+    const pluginNames = Object.keys(pluginTemplates)
+    const pluginTemplatesArray = Object.values(pluginTemplates)
+    for (let i = 0; i < pluginTemplatesArray.length; i++){
+        const name = pluginNames[i]
+        const template = pluginTemplatesArray[i]
+        if (answers.selectPlugins && answers.selectPlugins.includes(name)){
+            config.plugins![name] = { ...template, port: answers[`${name}Port`] }
         }
+    }
 
-        return selectedPlugins
+    config.ethereumPrivateKey = (answers.importPrivateKey) ? answers.importPrivateKey : Wallet.createRandom().privateKey
 
-    }
+    return config as Config
+}
 
-    async storeConfig(): Promise<string>{
-        const destinationFolder: string = await this.selectDestinationFolder()
-        // ensure the destination folder exists
-        if (!existsSync(destinationFolder)){
-            throw new Error(`Destination folder [${destinationFolder}] does not exist`)
-        }
-        const filename = `broker-config.json`
-        const finalPath = path.join(destinationFolder, filename)
-        // ask the user if they want to overwrite when the file already exists
-        if (existsSync(finalPath)){
-            const overwrite = await this.inquirerSinglePrompt({
-                type: 'confirm',
-                name: 'overwrite',
-                message: `Config file ${finalPath} already exists. Overwrite?`,
-                default: false
-            })
+export const selectDestinationPathPrompt = {
+    type: 'input',
+    name: 'selectDestinationPath',
+    message: `Select a path to store the generated config in `,
+    default: path.join(os.homedir(), '.streamr/broker-config.json'),
+    validate: (input: string, answers: inquirer.Answers = {}): string | boolean => {
+        try {
+            const filePath = input || answers.selectDestinationPath
+            const parentDirPath = filePath.substring(0, filePath.lastIndexOf('/'))
 
-            if (!overwrite) {
-                return this.storeConfig()
+            answers.parentDirPath = parentDirPath
```

Comment:
Do we need to store parentDirPath separately? If we already store the selectDestinationPath

Reply:
We do in order to create, if necessary, the folder on the later logic step

Reply:
Yes, we can calculate it there easily. If I understood correctly that selectDestinationPath is always path.dirname(selectDestinationPath), e.g. `/foo/bar/file.json` -> `/foo/bar/`